### PR TITLE
Initial Parser

### PR DESCRIPTION
**Pull request by andyafter**

### CODE REVIEW

File: merlin/json/metric_definitions.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[
```
Comment (kelindar) [inline merlin/json/metric_definitions.json]: Let's do YAML instead

**Review by kelindar — CHANGES_REQUESTED** [review]

**Review by kelindar — APPROVED** [review]

File: merlin/yaml/metric_definition.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+id: test_metric
```
Comment (ocassetti) [inline merlin/yaml/metric_definition.yml]: It is a list of metric definitions not just one

File: merlin/parser.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+from merlin.utils import LOGGER
+
+
+class MetricParser():
+    def __init__(self):
+        # the metric_fields is a set of metric field strings. This is for validation use.
+        self.metric_fields = set(['id', 'time', 'w_time', 'group_map', 'group_keys', 'func_map', 'func_expr', 'func_vars', 'compute_time', 'v_lvl', 'h_lvl', 'stages', 'description', 'version'])
+        return
+
+    def load_metrics(self, metric_db="merlin/yaml/metric_definition.yml"):
```
Comment (ocassetti) [inline merlin/parser.py]: I don't think we need any defaults for metric_db

File: merlin/parser.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+from merlin.metric import SourceMetric, Definition
+from merlin.stage import Stage
+from merlin.utils import LOGGER
+
+
+class MetricParser():
+    def __init__(self):
+        # the metric_fields is a set of metric field strings. This is for validation use.
+        self.metric_fields = set(['id', 'time', 'w_time', 'group_map', 'group_keys', 'func_map', 'func_expr', 'func_vars', 'compute_time', 'v_lvl', 'h_lvl', 'stages', 'description', 'version'])
```
Comment (ocassetti) [inline merlin/parser.py]: I think you can figure out the fields from SoruceMetric... 
In any case `group_map` `group_keys` `func_map` will not be part of the source metric or definition

**Review by ocassetti — APPROVED** [review]

**Pull request merged**